docs/source/user_guide/writing-extractors.rst

```diff
@@ -0,0 +1,179 @@
+      return "0.0.1"
+
+``get_data_output_category()``
+------------------------------
+
+This function should return a ``DataOutputCategory`` object, which tells MetaLad what kind of (meta)data it is dealing with. The following output categories are available in the enumeration ``datalad_metalad.extractors.base.DataOutputCategory``::
+
+ IMMEDIATE
+ FILE
+ DIRECTORY
+
+Output categories declare how the extractor delivers its results. If the output category is ``IMMEDIATE``, the result is returned by the ``extract()``-call. In case of ``FILE``, the extractor deposits the result in a file. This ie especially useful for extractors that are external programs. The category ``DIRECTORY`` is not yet supported. If you write an extractor in Python, you would usually use the output category ``IMMEDIATE`` and return extractor results from the ``extract()``-call.
+
+Example::
+  
+  def get_data_output_category(self) -> DataOutputCategory:
+      return DataOutputCategory.IMMEDIATE
+
+``get_required_content()``
+--------------------------
+
+This function is used in dataset-level extractors only.
+It will be called by MetaLad prior to metadata extraction.
+Its purpose is to allow the extractor to ensure that content that is required for metadata extraction is present
+(relevant, for example, if some of files to be inspected may be annexed).
+The function should return ``True`` if it has obtained the required content, or confirmed its presence.
+If it returns ``False``, metadata extraction will not proceed.
+
+This function can be a place to call ``dataset.get()``.
+It is advisable to disable result rendering (``result_renderer="disabled"``), because during metadata extraction, users will typically want to redirect standard output to a file or another command.
+
+Example::
+
+  def get_required_content(self) -> bool:
+      result = self.dataset.get("CITATION.cff", result_renderer="disabled")
+      return result[0]["status"] in ("ok", "notneeded")
+
+``is_content_required()``
+-------------------------
+
+This function is used in file-extractors only.
+It is a file-extractor counterpart to ``get_required_content()``.
+Its purpose is to tell MetaLad whether the file content is required or not
+(relevant for annexed files - extraction may depend on file content, or require only annex key).
+If the function returns ``True``, MetaLad will get the file content.
+If it returns ``False``, the get operation will not be performed.
+      
+``extract()``
+-------------
+
+This function is used for actual metadata extraction. It has one parameter called ``output_location``. If the output category of the extractor is ``DataOutputCategory.IMMEDIATE``, this parameter will be ``None``. If the output category is ``DataOutputCategory.FILE``. this parameter will contain either a file name or a ``file``-object into which the extractor can write its output.
+
+ The function should return an ``datalad_metalad.extractors.base.ExtractorResult`` object.
```

Comment:
```suggestion
The function should return an ``datalad_metalad.extractors.base.ExtractorResult`` object.
```

Looks like an "unexpected indent" made its way through and was caught by the CI.